app/presenters/search_result_presenter.rb

```diff
@@ -8,29 +8,84 @@ class SearchResultPresenter
            :es_score,
            to: :search_result
 
-  def initialize(search_result, metadata)
+  def initialize(search_result, metadata, doc_index, doc_count, finder_name, debug_score, highlight)
     @search_result = search_result
     @metadata = metadata
+    @index = doc_index + 1
+    @count = doc_count
+    @finder_name = finder_name
+    @debug_score = debug_score
+    @highlight = highlight
   end
 
   def to_hash
     {
-      title: title,
-      link: link,
-      summary: summary,
-      is_historic: is_historic,
-      government_name: government_name,
-      metadata: metadata,
-      show_metadata: show_metadata,
-      format: format,
-      es_score: es_score,
+      link: {
+        text: title,
+        path: link,
+        description: summary_text,
+        data_attributes: {
+          track_category: "navFinderLinkClicked",
+          track_action: "#{@finder_name}.#{@index}",
+          track_label: link,
+          track_options: {
+            dimension28: @count,
+            dimension29: title
+          }
+        }
+      },
+      metadata: structure_metadata,
+      metadata_raw: metadata,
+      subtext: subtext,
+      highlight: @highlight,
+      highlight_text: highlight_text
     }
   end
 
   def link
     search_result.path
   end
 
+  def structure_metadata
+    component_metadata = {}
+
+    if show_metadata
+      metadata.each do |meta|
```

Comment:
then this can become an each_with_object,

```
metadata.each_with_object({}) do |meta, component_metadata|
```

Reply:
I've learned more than one thing today! Changed, thanks!